eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -792,13 +794,19 @@ impWitsVKeyNeeded txBody = do
 data ImpTestEnv era = ImpTestEnv
   { iteFixup :: Tx era -> ImpTestM era (Tx era)
   , iteExpectLedgerRuleConformance ::
+      forall t.
+      Globals ->
       Either
         (NonEmpty (PredicateFailure (EraRule "LEDGER" era)))
         (State (EraRule "LEDGER" era), [Event (EraRule "LEDGER" era)]) ->
       LedgerEnv era ->
       LedgerState era ->
       Tx era ->
-      Expectation
+      ImpM t ()
+  -- ^ Note the use of higher ranked types here. This prevents the hook from
```

Comment:
Very nice!

Reply:
Unfortunately I had to revert it because the `ImpredicativeTypes` is broken in GHC 8 :/